pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -171,30 +171,34 @@ func (f *Fit) Filter(ctx context.Context, cycleState *framework.CycleState, pod
 	return nil
 }
 
-func getErrReason(rn v1.ResourceName) string {
-	return fmt.Sprintf("Insufficient %v", rn)
-}
-
 // InsufficientResource describes what kind of resource limit is hit and caused the pod to not fit the node.
 type InsufficientResource struct {
 	ResourceName v1.ResourceName
-	Requested    int64
-	Used         int64
-	Capacity     int64
+	// We explicitly have a parameter for reason to avoid formatting a message on the fly
+	// for common resources, which is expensive for cluster autoscaler simulations.
+	Reason    string
+	Requested int64
+	Used      int64
+	Capacity  int64
 }
 
 // Fits checks if node have enough resources to host the pod.
 func Fits(pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo, ignoredExtendedResources sets.String) []InsufficientResource {
-	return fitsRequest(computePodResourceRequest(pod), nodeInfo, ignoredExtendedResources)
+	r := fitsRequest(computePodResourceRequest(pod), nodeInfo, ignoredExtendedResources)
+	if len(r) == 0 {
```

Comment:
it doesn't look like we need this? Or is this some kind of hint to the GC?

Reply:
yeah, did that for the tests, but now I updated the tests to expect a non nil empty slice and reverted this.

Reply:
In general, tests should use `cmp.Diff`, so that wouldn't be an issue.

Reply:
cmp.diff didn't work either, nil and empty slice seems to return a difference.

Reply:
ack :(